pyresample/test/test_sgeom/test_arc.py

```diff
@@ -0,0 +1,312 @@
+        arc1 = SArc(SPoint.from_degrees(-180.0, -89.0),
+                    SPoint.from_degrees(-180.0, 89.0))
+        arc2 = SArc(SPoint.from_degrees(-171.03, -76.75),
+                    SPoint.from_degrees(170.34, -71.36))
+        arc1_orig = copy.deepcopy(arc1)
+        arc2_orig = copy.deepcopy(arc2)
+        point = arc1.intersection_point(arc2)
+        # Assert original arcs are unaffected
+        assert arc1_orig.end.lon == arc1.end.lon
+        assert arc2_orig.end.lon == arc2.end.lon
+        # Assert intersection result
+        assert point == SPoint.from_degrees(-180.0, -74.78847163)
+
+    def test_great_circles_equator_arcs_cases(self):
+        """Test behaviour when 2 arcs are lying around the equator."""
+        equator_arc1 = SArc(SPoint.from_degrees(0.0, 0.0), SPoint.from_degrees(50, 0.0))
+        equator_arc2 = SArc(SPoint.from_degrees(-50, 0.0), SPoint.from_degrees(0.0, 0.0))
+
+        # - If one arc is reversed, the results order is swapped
+        p1, p2 = equator_arc1._great_circle_intersections(equator_arc2)
+        assert np.allclose(p1.vertices_in_degrees, (0, 0))
+        assert np.allclose(p2.vertices_in_degrees, (-180, 0))
+
+        p1, p2 = equator_arc2._great_circle_intersections(equator_arc1)
+        assert np.allclose(p1.vertices_in_degrees, (0, 0))
+        assert np.allclose(p2.vertices_in_degrees, (-180, 0))
+
+        # - If both arc are reversed, results are the same
+        p1, p2 = equator_arc2.reverse_direction()._great_circle_intersections(equator_arc1.reverse_direction())
+        assert np.allclose(p1.vertices_in_degrees, (0, 0))
+        assert np.allclose(p2.vertices_in_degrees, (-180, 0))
+
+        # - Reversing the direction of one arc, lead inversion of points
+        p1, p2 = equator_arc2._great_circle_intersections(equator_arc1.reverse_direction())
+        assert np.allclose(p1.vertices_in_degrees, (-180, 0))
+        assert np.allclose(p2.vertices_in_degrees, (0, 0))
+
+    def test_intersection_equator_arcs_case(self):
+        """Test intersection point with 2 arcs lying around the equator."""
+        equator_arc1 = SArc(SPoint.from_degrees(0.0, 0.0), SPoint.from_degrees(50, 0.0))
+        equator_arc2 = SArc(SPoint.from_degrees(-50, 0.0), SPoint.from_degrees(0.0, 0.0))
+        equator_arc3 = SArc(SPoint.from_degrees(-50, 0.0), SPoint.from_degrees(-20, 0.0))
+
+        # Touching arcs
+        p = equator_arc2.intersection_point(equator_arc1)
+        assert np.allclose(p.vertices_in_degrees, (0, 0))
+
+        p = equator_arc2.intersection_point(equator_arc1.reverse_direction())
+        assert np.allclose(p.vertices_in_degrees, (0, 0))
+
+        # Overlapping arcs
+        p = equator_arc2.intersection_point(equator_arc3)
+        assert isinstance(p, type(None))
+
+        # Disjoint arcs
+        p = equator_arc3.intersection_point(equator_arc1)
+        assert isinstance(p, type(None))
```

Comment:
Same here. These seem like separate tests. Perhaps a separate class or fixtures if they need to reuse the same input data?